### PR TITLE
make pandoc optional

### PR DESCRIPTION
**Pull request by bergus**
Cuts down my build size from 41,005,585 Bytes to 31,203,857 Bytes.

### CODE REVIEW

File: hakyll.cabal [hunk 1]
```diff
@@ -99,6 +99,10 @@ Flag buildWebsite
   Default:     False
   Manual:      True
 
+Flag usePandoc
+  Description: Include Pandoc support
+  Default:     True
```
Comment (jaspervdj) [inline hakyll.cabal]: We should also make this a `Manual: True` flag

File: tests/Hakyll/Web/Template/Tests.hs [hunk 2]
```diff
@@ -89,7 +88,9 @@ test (outf, tplf, itemf) = do
     out  <- resourceString provider outf
     tpl  <- testCompilerDone store provider tplf templateBodyCompiler
     item <- testCompilerDone store provider itemf $
-        pandocCompiler >>= applyTemplate (itemBody tpl) testContext
+        getResourceBody
+        >>= renderParagraphs
```
Comment (jaspervdj) [inline tests/Hakyll/Web/Template/Tests.hs]: I would put a comment here about why we're not using pandoc

**Comment by bergus** [comment]
OK, updated.

**Comment by jaspervdj** [comment]
Thanks again!

**Comment by bergus** [comment]
What is weird though is that I'm getting a warning
```
$ stack install
hakyll-4.12.0.1: build (lib + exe)

Warning: The following modules should be added to exposed-modules or other-modules in C:\Users\…\hakyll\hakyll.cabal:
             - In hakyll-tests component:
                 Hakyll.Web.Pandoc.FileType.Tests

         Missing modules in the cabal file are likely to cause undefined reference errors from the linker, along with other problems.
hakyll-4.12.0.1: copy/register
```
when I run my project with `usePandoc: False`

**Comment by jaspervdj** [comment]
I guess `Hakyll.Web.Pandoc.FileType` doesn't _really_ depend on pandoc.  I'll fix this.

**Pull request merged**